pimd/pim_zebra.c

```diff
@@ -394,133 +394,105 @@ static int pim_zebra_if_address_del(int command, struct zclient *client,
 
-				nbr = pim_neighbor_find(
-					old.source_nexthop.interface,
-					old.rpf_addr.u.prefix4);
-				if (nbr)
-					pim_jp_agg_remove_group(
-						nbr->upstream_jp_agg, up);
+			if (up->join_state == PIM_UPSTREAM_JOINED) {
+				/*
+				 * If we come up real fast we can be here
+				 * where the mroute has not been installed
+				 * so install it.
+				 */
+				if (!up->channel_oil->installed)
+					pim_mroute_add(up->channel_oil,
+						       __PRETTY_FUNCTION__);
 
 				/*
-				 * We have detected a case where we might need
-				 * to rescan
-				 * the inherited o_list so do it.
+				 * RFC 4601: 4.5.7.  Sending (S,G)
+				 * Join/Prune Messages
+				 *
+				 * Transitions from Joined State
+				 *
+				 * RPF'(S,G) changes not due to an Assert
+				 *
+				 * The upstream (S,G) state machine remains
+				 * in Joined state. Send Join(S,G) to the new
+				 * upstream neighbor, which is the new value
+				 * of RPF'(S,G).  Send Prune(S,G) to the old
+				 * upstream neighbor, which is the old value
+				 * of RPF'(S,G).  Set the Join Timer (JT) to
+				 * expire after t_periodic seconds.
 				 */
-				if (up->channel_oil->oil_inherited_rescan) {
-					pim_upstream_inherited_olist_decide(pim,
-									    up);
-					up->channel_oil->oil_inherited_rescan =
-						0;
-				}
+				pim_jp_agg_switch_interface(&old, &up->rpf, up);
 
-				if (up->join_state == PIM_UPSTREAM_JOINED) {
-					/*
-					 * If we come up real fast we can be
-					 * here
-					 * where the mroute has not been
-					 * installed
-					 * so install it.
-					 */
-					if (!up->channel_oil->installed)
-						pim_mroute_add(
-							up->channel_oil,
-							__PRETTY_FUNCTION__);
-
-					/*
-					 * RFC 4601: 4.5.7.  Sending (S,G)
-					 * Join/Prune Messages
-					 *
-					 * Transitions from Joined State
-					 *
-					 * RPF'(S,G) changes not due to an
-					 * Assert
-					 *
-					 * The upstream (S,G) state machine
-					 * remains in Joined
-					 * state. Send Join(S,G) to the new
-					 * upstream neighbor, which is
-					 * the new value of RPF'(S,G).  Send
-					 * Prune(S,G) to the old
-					 * upstream neighbor, which is the old
-					 * value of RPF'(S,G).  Set
-					 * the Join Timer (JT) to expire after
-					 * t_periodic seconds.
-					 */
-					pim_jp_agg_switch_interface(
-						&old, &up->rpf, up);
-
-					pim_upstream_join_timer_restart(up,
-									&old);
-				} /* up->join_state == PIM_UPSTREAM_JOINED */
-
-				/* FIXME can join_desired actually be changed by
-				   pim_rpf_update()
-				   returning PIM_RPF_CHANGED ? */
-				pim_upstream_update_join_desired(pim, up);
-
-			} /* PIM_RPF_CHANGED */
-
-		} /* for (qpim_upstream_list) */
-	}
+				pim_upstream_join_timer_restart(up, &old);
+			} /* up->join_state == PIM_UPSTREAM_JOINED */
 
-	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
-		pim = vrf->info;
-		if (!pim)
-			continue;
+			/* FIXME can join_desired actually be changed by
+			   pim_rpf_update()
+			   returning PIM_RPF_CHANGED ? */
```

Comment:
What's your plan for this "FIXME"?

Reply:
This FIXME is in the above deleted code.  I just moved it around a bit.  I would have to rewrap my head around this code comment from a long time ago to figure this out.

Reply:
Okay, just let me know how you think we should proceed, e.g., fix now or open an issue to track what needs to be done...

Reply:
I am not taking the time to fix this issue now.  It's been there over a year currently. I'll file an issue.